rm-datasource/src/main/java/io/seata/rm/datasource/undo/postgresql/keyword/PostgresqlKeywordChecker.java

```diff
@@ -0,0 +1,533 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.rm.datasource.undo.postgresql.keyword;
+
+import io.seata.rm.datasource.undo.KeywordChecker;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * The type oracle undo delete executor.
+ *
+ * @author japsercloud
+ * @date 2019/09/11
+ */
+public class PostgresqlKeywordChecker implements KeywordChecker {
+    private static volatile KeywordChecker keywordChecker = null;
+    private static volatile Set<String> keywordSet = null;
+
+    private PostgresqlKeywordChecker() {
+    }
+
+    /**
+     * get instance of type oracle keyword checker
+     *
+     * @return instance
+     */
+    public static KeywordChecker getInstance() {
+        if (keywordChecker == null) {
+            synchronized (PostgresqlKeywordChecker.class) {
+                if (keywordChecker == null) {
+                    keywordChecker = new PostgresqlKeywordChecker();
+                    keywordSet = Arrays.stream(OracleKeyword.values()).map(OracleKeyword::name).collect(Collectors.toSet());
+                }
+            }
+        }
+        return keywordChecker;
+    }
+
+    /**
+     * oracle keyword
+     */
+    private enum OracleKeyword {
```

Comment:
OracleKeyword?

Reply:
This is oracle keyword. please change to postgre sql keyword.

Reply:
changed